### PR TITLE
Refactor reporter.js

### PR DESCRIPTION
**Pull request by netmajor**
Please check it ASAP. Because after each change in reporter.js i have to apply that changes in my branch :/

### CODE REVIEW

**Comment by BorisOsipov** [comment]
@netmajor for what this refactoring needs? what advantages it will brings to the project codebase?

**Comment by netmajor** [comment]
@BorisOsipov If you look into https://codeclimate.com/github/webdriverio/wdio-allure-reporter/issues , you see that constructor contains too many code. Good practise tells that it should be refactor. 
In last days when me and others adds new allure api to this class, we can see that it start to grow...

Maybe it now is not so big but that refactor fix this problem for future.

**Comment by BorisOsipov** [comment]
>constructor contains too many code. Good practise tells that it should be refactor.

It is good practise when you have constructor to create big\compelex object and it suggests to use builders and other object construction patterns for it.
In this case you just move logic from one method to another without any advantages. You don't add unit tests for them. Just refactoring for refactoring...

So and don't tell it doesn't need, but I don't respect changes to appease static code analyse tools.

**Comment by netmajor** [comment]
I think that all this logic for events shouldn't exist in constructor, so i move it to functions. It is easier to read contractor then.

File: lib/reporter.js [hunk 1]
```diff
@@ -25,150 +25,199 @@ class AllureReporter extends events.EventEmitter {
...
+        this.testPending = this.testPending.bind(this)
+
+        this.runnerCommand = this.runnerCommand.bind(this)
+        this.runnerResult = this.runnerResult.bind(this)
+        this.runnerScreenshot = this.runnerScreenshot.bind(this)
+
+        this.allureFeature = this.allureFeature.bind(this)
+        this.allureAddEnvironment = this.allureAddEnvironment.bind(this)
+        this.allureAttachment = this.allureAttachment.bind(this)
+        this.allureStory = this.allureStory.bind(this)
```
Comment (christian-bromann) [inline lib/reporter.js]: If you do:
```js
this.on('test:pass', ::this.testPass)
```
you can remove the `bindEventFunctions` method completely

**Comment by netmajor** [comment]
I need to create new fork, because last commit 'AddDescription Allure API' has other number on fork than master and I have problem with pulling changes to my fork :/

**Pull request closed without merging**